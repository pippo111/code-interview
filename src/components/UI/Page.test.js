/* global describe, test, it, expect */
import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import { Page } from '.'

const ChildComponent = () => <div>Hello leo!</div>

describe('Page', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<Page><div>Hello</div></Page>)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders page title with child component', () => {
    const pageTitle = 'Test title'

    const component = mount(
      <Page title={pageTitle}>
        <ChildComponent />
      </Page>
    )

    expect(component.find('h1').text()).toEqual(pageTitle)
    expect(component.find(ChildComponent).length).toEqual(1)
  })
})
