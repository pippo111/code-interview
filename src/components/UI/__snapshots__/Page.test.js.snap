// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page snapshot renders 1`] = `
<div
  className="sc-AykKC gUNwOj"
>
  <div>
    Hello
  </div>
</div>
`;
